plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -32,10 +33,21 @@ def __init__(
             name: Name of the task
             query: DuckDB query to execute
             inputs: The query parameters to be used while executing the query
+            motherduck_secret: Secret containing a MotherDuck authentication token
         """
+        if motherduck_secret is None:
+            # create an in-memory database that's non-persistent
+            self._con = duckdb.connect(":memory:")
+        else:
+            # connect to motherduck
+            motherduck_token = current_context().secrets.get(
+                group=motherduck_secret.group,
+                key=motherduck_secret.key,
+                group_version=motherduck_secret.group_version,
+            )
+            self._con = duckdb.connect("md:", config={"motherduck_token": motherduck_token})
```

Comment:
My concern is that this is hardcoded for motherduck. If there was a non-motherduck provider for duckdb, then we'll need to add another if statement & `another_provider_sercret` to configure them.

The only way I see doing this is to provide a callable:

```python
def motherduck_connect(token):
    return duckdb.connect("md:", config={"motherduck_token": token})

def memory_connect(token):
    return duckdb.connect(":memory:")

class DuckDBQuery:
    def __init__(self, ..., construct_duckdb_connect=motherduck_connect):
        if secret is None:
            self._con = construct_duckdb_connect(None)
        else:
            token = current_conect()...
            self._con = construct_duckdb_connect(token)
```

The above API also makes the rest of `duckdb.connect` configurable to the user.

Reply:
good point. let me make this a bit more general